### DISCUSSION TITLE
how to add recipes to withered beacon?

### DISCUSSION THREAD

**Discussion by entity1O1** [nonamecrackers2/crackers-wither-storm-mod#1919] [Q&A] [answered]
can you add recipes to withered beacon yet? I want to use it in a modpack I am making, are there future plans for this?

**Comment by nonamecrackers2** [comment, marked as answer]
You can, you need to have a datapack with your recipes in ``data/<namespace>/recipes``, then include a recipe JSON similiar to the following:

For a mob:

    {
      "type": "witherstormmod:resummon_super_beacon",
      "condition": "all_supports",
      "entity": "witherstormmod:sickened_pillager",
      "ingredients": [
        {
          "item": "witherstormmod:withered_flesh"
        },
        {
          "item": "witherstormmod:withered_flesh"
        },
        {
          "item": "witherstormmod:withered_flesh"
        },
        {
          "item": "minecraft:crossbow"
        },
        {
          "item": "minecraft:arrow"
        },
        {
          "item": "minecraft:arrow"
        }
      ]
    }

For an item:

    {
      "type": "witherstormmod:item_craft_super_beacon",
      "condition": "main_activated",
      "ingredients": [
        {
          "item": "minecraft:rotten_flesh"
        },
        {
          "item": "minecraft:rotten_flesh"
        },
        {
          "item": "minecraft:rotten_flesh"
        },
        {
          "item": "minecraft:rotten_flesh"
        },
        {
          "item": "minecraft:rotten_flesh"
        },
        {
          "item": "minecraft:rotten_flesh"
        }
      ],
      "result": {
        "item": "witherstormmod:withered_flesh"
      }
    }

**Reply by MCXPStudios** [reply]
What options of conditions are available, if I may ask?

**Reply by nonamecrackers2** [reply]
For sure! Here's the conditions:

- ``none``: No conditions
- ``main_activated``: The main beacon is at least activated
- ``all_supports``: The main beacon and the supports are activated
- ``fully_completed``: The main beacon and the supports are activated, and the beacon is fully leveled

**Reply by MCXPStudios** [reply]
> For sure! Here's the conditions:
> 
> * `none`: No conditions
> * `main_activated`: The main beacon is at least activated
> * `all_supports`: The main beacon and the supports are activated
> * `fully_completed`: The main beacon and the supports are activated, and the beacon is fully leveled

Thank you :)
